Keymap editor

ejaquay opened the pull request:
Working keymap editor.   Hope it's use is intuitive enough.  Otherwise please suggest how to improve / document.

Honestly I initially thought the effort of creating a GUI keymap editor was overkill because users could just edit a custom.keymap file using notepad.   They still can but now that the GUI is working I see that I was wrong.  Once I became used to it the GUI editor is great for on the fly key changes and that is likely how it will be used.  When user gets tired of reaching for a key combo or remembering what each keymap does they can make a custom key map fix as they discover their issues.  In my world I already have the numpad keys remapped to make Nitros-9 command line editing easier for me.

Commit:
Add comment to explain the CoCo row / column codes in translation tables

Commit:
Merge branch 'VCC_2.0.1c' of github.com:ejaquay/VCC into VCC_2.0.1c

Commit:
Phase 1 of custom keymap editor.

Fix size of custom keymap table (keyboardLayout.[ch])
Include keyboardEdit.c in project (Vcc.vcxproj)
Load custom keymap from file (config.[ch] keyboardEdit.[ch])
Partial keymap edit dialog (Vcc.rc,resource.h,keyboardEdit.[ch]
Git ignore vim swap files (.gitignore)

Commit:
Phase 1 of custom keymap editor.

Fix size of custom keymap table (keyboardLayout.[ch])
Include keyboardEdit.c in project (Vcc.vcxproj)
Load custom keymap from file (config.[ch] keyboardEdit.[ch])
Partial keymap edit dialog (Vcc.rc,resource.h,keyboardEdit.[ch]
Git ignore vim swap files (.gitignore)

Commit:
Merge branch 'editkeymap' of github.com:ejaquay/VCC into editkeymap

Commit:
Enable custom key map editing

Key map editer now actually remaps keys
Still to do:  Load and Save function
Vcc.c OEMscan corrected for arrow keys.

Commit:
Disable set button after key map is changed

Commit:
Limited save keymap functionality added

Save keymap button saves keymap

Sitll TODO:
      Ability to wite to a different file.
      Load keymap function.

Commit:
Fix not using correct keymap file name

Commit:
Disable buttons until needed, more error checking

Additional save keymap error checking.
Disable save keymap button until changes made.

Commit:
'Save Keymap' and 'Select File' buttons now working

Still TODO:  Save selected file in vcc init.  Currently Vcc defaults to
custom.keymap on startup, requiring user to use "Select File" if they
want to use some other file.

Commit:
Move keynames to header file

Commit:
Save keymap filename to ini file

o Select button in keymap edit dialog now saves keymap file path to vcc ini.

o set keymap path from vcc ini on startup.

o Improve text in edit dialog.

o Add Index for scancode table lookups.

o Misc code cleanup and bug fixes.

Warning: It is likely there are still yet to be found bugs.

BGPierce commented:
I'll get it merged in as soon as I can. We have a large storm coming in today and I know internet access is going to go nuts as the whole county went back to virtual schools for the day to keep busses off the roads, and the local IPs will be bogged down, so it may be this weekend.
Actually, I saw the need for this long ago, hence the editor GUI that has sit idle in the "Resources" for quite awhile. :-)


-----Original Message-----
From: Ed Jaquay ***@***.***>
To: VCCE/VCC ***@***.***>
Cc: Subscribed ***@***.***>
Sent: Wed, Mar 17, 2021 5:10 pm
Subject: [VCCE/VCC] Keymap editor (#76)

Working keymap editor. Hope it's use is intuitive enough. Otherwise please suggest how to improve / document.Honestly I initially thought the effort of creating a GUI keymap editor was overkill because users could just edit a custom.keymap file using notepad. They still can but now that the GUI is working I see that I was wrong. Once I became used to it the GUI editor is great for on the fly key changes and that is likely how it will be used. When user gets tired of reaching for a key combo or remembering what each keymap does they can make a custom key map fix as they discover their issues. In my world I already have the numpad keys remapped to make Nitros-9 command line editing easier for me.
You can view, comment on, or merge this pull request online at:
  https://github.com/VCCE/VCC/pull/76
Commit Summary
   
   - Add comment to explain the CoCo row / column codes in translation tables
   - Merge branch 'VCC_2.0.1c' of github.com:ejaquay/VCC into VCC_2.0.1c
   - Phase 1 of custom keymap editor.
   - Phase 1 of custom keymap editor.
   - Merge branch 'editkeymap' of github.com:ejaquay/VCC into editkeymap
   - Enable custom key map editing
   - Disable set button after key map is changed
   - Limited save keymap functionality added
   - Fix not using correct keymap file name
   - Disable buttons until needed, more error checking
   - 'Save Keymap' and 'Select File' buttons now working
   - Move keynames to header file
   - Save keymap filename to ini file

File Changes
   
   -  M .gitignore (1) 
   -  M Vcc.c (11) 
   -  M Vcc.rc (285) 
   -  M Vcc.vcxproj (4) 
   -  M config.c (82) 
   -  M config.h (7) 
   -  A docs/custom.keymap (137) 
   -  M keyboard.c (2) 
   -  A keyboardEdit.c (1055) 
   -  A keyboardEdit.h (47) 
   -  M keyboardLayout.c (20) 
   -  M keyboardLayout.h (2) 
   -  A keynames.h (271) 
   -  M resource.h (284) 

Patch Links:
   
   - https://github.com/VCCE/VCC/pull/76.patch
   - https://github.com/VCCE/VCC/pull/76.diff
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

Commit:
Use _BitScanForward intrinsic to optimize CoCo Key lookup

Commit:
Add key Map Editor documentation

File docs/KeyMapEditor.txt briefly describes the editor operation

Also bug where the keymap modified status was lost when leaving the
key map editor has been fixed.

ejaquay commented:
I added file docs/KeyMapEditor.txt to describe editor operation
also fixed a minor bug and added a table lookup optimization.

Commit:
Some documentation Fixes

Commit:
Notify user if Custom keymap not selected when keyboard config EDIT clicked.
